RFS/src/main/java/com/rfs/common/S3SnapshotCreator.java

```diff
@@ -0,0 +1,152 @@
+package com.rfs.common;
+
+import java.net.HttpURLConnection;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class S3SnapshotCreator {
+    private static final Logger logger = LogManager.getLogger(S3SnapshotCreator.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final ConnectionDetails connectionDetails;
+    private final String snapshotName;
+    private final String s3Uri;
+    private final String s3Region;
+
+    public S3SnapshotCreator(String snapshotName, ConnectionDetails connectionDetails, String s3Uri, String s3Region) {
+        this.snapshotName = snapshotName;
+        this.connectionDetails = connectionDetails;
+        this.s3Uri = s3Uri;
+        this.s3Region = s3Region;
+    }
+
+    public String getSnapshotName() {
+        return snapshotName;
+    }
+
+    /*
+     * Extracts the bucket name from the S3 URI
+     * s3://my-bucket-name/my-folder/my-nested-folder => my-bucket-name
+     */
+    public String getBucketName() {
+        return s3Uri.split("/")[2];
+    }
+
+    /*
+     * Extracts the base path from the S3 URI; could be nested arbitrarily deep
+     * s3://my-bucket-name/my-folder/my-nested-folder => my-folder/my-nested-folder
+     */
+    public String getBasePath() {
```

Comment:
Nice to have some validation around the URI at some point, assuming if someone accidentally passed an http s3 link for the `s3Uri` this might produce a weird error

Reply:
+1.  There's a lot of stuff like that we'll need to tackle before handing the code out to users "for real".  I figured that would naturally follow from adding unit tests, but the code's shape has been changing so rapidly that I haven't thought writing those had made sense so far.  We should keep a close eye on when we think the shape is stabilizing.